src/upload.rs

```diff
@@ -133,7 +140,12 @@ impl SnapshotUploader {
             // The file length may not be an exact multiple of the block size,
             // so we need to keep track of how many bytes are left for the call
             // to `read_exact` later.
-            let data_length = cmp::min(block_size, remaining_data);
+            let data_length = cmp::min(i64::from(block_size), remaining_data);
+            let data_length = i32::try_from(data_length).context(error::ConvertNumberSnafu {
+                what: "data length",
+                number: data_length.to_string(),
+                target: "i32",
+            })?;
```

Comment:
Why do we need to convert an i64 to an i32 and then to a usize, instead of converting it directly to a usize?

Reply:
i32 is probably unnecessary since we're guaranteed the size will be inside a 32 bit int. I'll remove that conversion.